warning: in the working copy of 'Untitled.ipynb', LF will be replaced by CRLF the next time Git touches it
diff --git a/Untitled.ipynb b/Untitled.ipynb
index 089b497..ced5a59 100644
--- a/Untitled.ipynb
+++ b/Untitled.ipynb
@@ -12,17 +12,17 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 4,
    "id": "e610e5ec-b382-4d7a-969f-783a8a4858bd",
    "metadata": {},
    "outputs": [],
    "source": [
-    "df = pd.read_parquet(r'C:\\Users\\eelil\\OneDrive\\Desktop\\Capstone\\Machine_Learning_Analysis_of_Bank_Fraud\\data\\cust_loc.parquet')"
+    "df = pd.read_parquet(r'C:\\Users\\eelil\\OneDrive\\Desktop\\Capstone\\Machine_Learning_Analysis_of_Bank_Fraud\\cust_loc.parquet')"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 3,
+   "execution_count": 5,
    "id": "33b67860-7b11-4afd-9a48-a0cf858bb5ef",
    "metadata": {},
    "outputs": [
@@ -47,162 +47,75 @@
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
-       "      <th>trans_dt</th>\n",
-       "      <th>cc_num</th>\n",
-       "      <th>merchant</th>\n",
-       "      <th>category</th>\n",
-       "      <th>amt</th>\n",
        "      <th>city</th>\n",
        "      <th>state</th>\n",
-       "      <th>lat</th>\n",
-       "      <th>long</th>\n",
-       "      <th>job</th>\n",
-       "      <th>dob</th>\n",
-       "      <th>trans_num</th>\n",
-       "      <th>merch_lat</th>\n",
-       "      <th>merch_long</th>\n",
-       "      <th>is_fraud</th>\n",
-       "      <th>trans_type</th>\n",
+       "      <th>lat_long</th>\n",
+       "      <th>cust_lat</th>\n",
+       "      <th>cust_long</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0</th>\n",
-       "      <td>2019-01-01 00:00:18</td>\n",
-       "      <td>2703186189652095</td>\n",
-       "      <td>fraud_Rippin, Kub and Mann</td>\n",
-       "      <td>misc_net</td>\n",
-       "      <td>4.97</td>\n",
        "      <td>Moravian Falls</td>\n",
        "      <td>NC</td>\n",
-       "      <td>36.0788</td>\n",
-       "      <td>-81.1781</td>\n",
-       "      <td>Psychologist, counselling</td>\n",
-       "      <td>1988-03-09</td>\n",
-       "      <td>0b242abb623afc578575680df30655b9</td>\n",
-       "      <td>36.011293</td>\n",
-       "      <td>-82.048315</td>\n",
-       "      <td>0</td>\n",
-       "      <td>1</td>\n",
+       "      <td>[36.102686500000004, -81.18158238008564]</td>\n",
+       "      <td>36.102687</td>\n",
+       "      <td>-81.181582</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1</th>\n",
-       "      <td>2019-01-01 00:00:44</td>\n",
-       "      <td>630423337322</td>\n",
-       "      <td>fraud_Heller, Gutmann and Zieme</td>\n",
-       "      <td>grocery_pos</td>\n",
-       "      <td>107.23</td>\n",
        "      <td>Orient</td>\n",
        "      <td>WA</td>\n",
-       "      <td>48.8878</td>\n",
-       "      <td>-118.2105</td>\n",
-       "      <td>Special educational needs teacher</td>\n",
-       "      <td>1978-06-21</td>\n",
-       "      <td>1f76529f8574734946361c461b024d99</td>\n",
-       "      <td>49.159047</td>\n",
-       "      <td>-118.186462</td>\n",
-       "      <td>0</td>\n",
-       "      <td>2</td>\n",
+       "      <td>[48.8660142, -118.2027814]</td>\n",
+       "      <td>48.866014</td>\n",
+       "      <td>-118.202781</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2</th>\n",
-       "      <td>2019-01-01 00:00:51</td>\n",
-       "      <td>38859492057661</td>\n",
-       "      <td>fraud_Lind-Buckridge</td>\n",
-       "      <td>entertainment</td>\n",
-       "      <td>220.11</td>\n",
        "      <td>Malad City</td>\n",
        "      <td>ID</td>\n",
-       "      <td>42.1808</td>\n",
-       "      <td>-112.2620</td>\n",
-       "      <td>Nature conservation officer</td>\n",
-       "      <td>1962-01-19</td>\n",
-       "      <td>a1a22d70485983eac12b5b88dad1cf95</td>\n",
-       "      <td>43.150704</td>\n",
-       "      <td>-112.154481</td>\n",
-       "      <td>0</td>\n",
-       "      <td>0</td>\n",
+       "      <td>[42.1915872, -112.250798]</td>\n",
+       "      <td>42.191587</td>\n",
+       "      <td>-112.250798</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3</th>\n",
-       "      <td>2019-01-01 00:01:16</td>\n",
-       "      <td>3534093764340240</td>\n",
-       "      <td>fraud_Kutch, Hermiston and Farrell</td>\n",
-       "      <td>gas_transport</td>\n",
-       "      <td>45.00</td>\n",
        "      <td>Boulder</td>\n",
        "      <td>MT</td>\n",
-       "      <td>46.2306</td>\n",
-       "      <td>-112.1138</td>\n",
-       "      <td>Patent attorney</td>\n",
-       "      <td>1967-01-12</td>\n",
-       "      <td>6b849c168bdad6f867558c3793159a81</td>\n",
-       "      <td>47.034331</td>\n",
-       "      <td>-112.561071</td>\n",
-       "      <td>0</td>\n",
-       "      <td>2</td>\n",
+       "      <td>[46.236595, -112.120834]</td>\n",
+       "      <td>46.236595</td>\n",
+       "      <td>-112.120834</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4</th>\n",
-       "      <td>2019-01-01 00:03:06</td>\n",
-       "      <td>375534208663984</td>\n",
-       "      <td>fraud_Keeling-Crist</td>\n",
-       "      <td>misc_pos</td>\n",
-       "      <td>41.96</td>\n",
        "      <td>Doe Hill</td>\n",
        "      <td>VA</td>\n",
-       "      <td>38.4207</td>\n",
-       "      <td>-79.4629</td>\n",
-       "      <td>Dance movement psychotherapist</td>\n",
-       "      <td>1986-03-28</td>\n",
-       "      <td>a41d7549acf90789359a9aa5346dcb46</td>\n",
-       "      <td>38.674999</td>\n",
-       "      <td>-78.632459</td>\n",
-       "      <td>0</td>\n",
-       "      <td>2</td>\n",
+       "      <td>[38.4320658, -79.4444882]</td>\n",
+       "      <td>38.432066</td>\n",
+       "      <td>-79.444488</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "text/plain": [
-       "             trans_dt            cc_num                            merchant  \\\n",
-       "0 2019-01-01 00:00:18  2703186189652095          fraud_Rippin, Kub and Mann   \n",
-       "1 2019-01-01 00:00:44      630423337322     fraud_Heller, Gutmann and Zieme   \n",
-       "2 2019-01-01 00:00:51    38859492057661                fraud_Lind-Buckridge   \n",
-       "3 2019-01-01 00:01:16  3534093764340240  fraud_Kutch, Hermiston and Farrell   \n",
-       "4 2019-01-01 00:03:06   375534208663984                 fraud_Keeling-Crist   \n",
+       "             city state                                  lat_long   cust_lat  \\\n",
+       "0  Moravian Falls    NC  [36.102686500000004, -81.18158238008564]  36.102687   \n",
+       "1          Orient    WA                [48.8660142, -118.2027814]  48.866014   \n",
+       "2      Malad City    ID                 [42.1915872, -112.250798]  42.191587   \n",
+       "3         Boulder    MT                  [46.236595, -112.120834]  46.236595   \n",
+       "4        Doe Hill    VA                 [38.4320658, -79.4444882]  38.432066   \n",
        "\n",
-       "        category     amt            city state      lat      long  \\\n",
-       "0       misc_net    4.97  Moravian Falls    NC  36.0788  -81.1781   \n",
-       "1    grocery_pos  107.23          Orient    WA  48.8878 -118.2105   \n",
-       "2  entertainment  220.11      Malad City    ID  42.1808 -112.2620   \n",
-       "3  gas_transport   45.00         Boulder    MT  46.2306 -112.1138   \n",
-       "4       misc_pos   41.96        Doe Hill    VA  38.4207  -79.4629   \n",
-       "\n",
-       "                                 job        dob  \\\n",
-       "0          Psychologist, counselling 1988-03-09   \n",
-       "1  Special educational needs teacher 1978-06-21   \n",
-       "2        Nature conservation officer 1962-01-19   \n",
-       "3                    Patent attorney 1967-01-12   \n",
-       "4     Dance movement psychotherapist 1986-03-28   \n",
-       "\n",
-       "                          trans_num  merch_lat  merch_long  is_fraud  \\\n",
-       "0  0b242abb623afc578575680df30655b9  36.011293  -82.048315         0   \n",
-       "1  1f76529f8574734946361c461b024d99  49.159047 -118.186462         0   \n",
-       "2  a1a22d70485983eac12b5b88dad1cf95  43.150704 -112.154481         0   \n",
-       "3  6b849c168bdad6f867558c3793159a81  47.034331 -112.561071         0   \n",
-       "4  a41d7549acf90789359a9aa5346dcb46  38.674999  -78.632459         0   \n",
-       "\n",
-       "   trans_type  \n",
-       "0           1  \n",
-       "1           2  \n",
-       "2           0  \n",
-       "3           2  \n",
-       "4           2  "
+       "    cust_long  \n",
+       "0  -81.181582  \n",
+       "1 -118.202781  \n",
+       "2 -112.250798  \n",
+       "3 -112.120834  \n",
+       "4  -79.444488  "
       ]
      },
-     "execution_count": 3,
+     "execution_count": 5,
      "metadata": {},
      "output_type": "execute_result"
     }
diff --git a/data/cust_loc_map.py b/data/cust_loc_map.py
index 8c6dd14..38a1e9e 100644
--- a/data/cust_loc_map.py
+++ b/data/cust_loc_map.py
@@ -4,9 +4,10 @@ import wandb
 import os
 import wandb
 from geopy.geocoders import Nominatim
+from time import time,sleep
 
 logging.basicConfig(
-    level=logging.INFO, format="%(asctime)-20s %(message)s", filemode="w"
+    level=logging.INFO, format="%(asctime)-20s %(message)s", filemode="a"
 )
 logger = logging.getLogger()
 
@@ -34,13 +35,16 @@ def get_lat_long(city, state):
         location = geolocator.geocode(f"{city}, {state}")
         if location:
             return location.latitude, location.longitude
+
         else:
             return None, None
     except Exception as e:
         logger.error(f"Error fetching location for {city}, {state}: {e}")
         return None, None
+    finally:
+        sleep(1)
 
-
+logger.info('Getting the geocoding file')
 # apply geocoding
 cust_loc["lat_long"] = cust_loc.apply(
     lambda x: get_lat_long(x["city"], x["state"]), axis=1
diff --git a/data/data.py b/data/data.py
index edb426e..357a5fe 100644
--- a/data/data.py
+++ b/data/data.py
@@ -6,7 +6,7 @@ import wandb
 
 # setting up logger
 logging.basicConfig(
-    level=logging.INFO, format="%(asctime)-20s %(message)s", filemode="w"
+    level=logging.INFO, format="%(asctime)-20s %(message)s", filemode="a"
 )
 logger = logging.getLogger()
 
diff --git a/feature_creation.py b/feature_creation.py
index 91acffd..0d6baa8 100644
--- a/feature_creation.py
+++ b/feature_creation.py
@@ -7,25 +7,13 @@ from time import time, sleep
 from haversine import haversine, Unit
 import swifter
 from rapidfuzz import fuzz, process
-from geopy.geocoders import Nominatim
+
 
 logging.basicConfig(
-    level=logging.INFO, format="%(asctime)-20s %(message)s", filemode="w"
+    level=logging.INFO, format="%(asctime)-20s %(message)s", filemode="a"
 )
 logger = logging.getLogger()
 
-geolocator = Nominatim(user_agent="geoapi")
-
-def get_lat_long(city=None, state=None):
-    try:
-        location = geolocator.geocode(f"{city}, {state}")
-        if location:
-            return location.latitude, location.longitude
-        else:
-            return None, None
-    except Exception as e:
-        logger.error(f"Error fetching location for {city}, {state}: {e}")
-        return None, None
 
 def feature_creation():
     # starting logging
@@ -80,21 +68,11 @@ def feature_creation():
 
     # accessing pre-mapped customer lat and long values
     # this file was created in Feature_Engineering_Test.ipynb
-    #FIXME: Logger info will need to be corrected if rest of file works
     logger.info(
         "Loading pre-mapped lat and long coordinates for customer's city and state, and using it to create 'cust_lat' and 'cust_long' columns."
     )
-    #cust_loc = pd.read_parquet(r"data/cust_loc.parquet")
-    #FIXME: geting lat and long
-    cust_loc = df[["city", "state"]].drop_duplicates()
-
-    cust_loc["lat_long"] = cust_loc.apply(
-        lambda x: get_lat_long(x["city"], x["state"]), axis=1
-    )
-    cust_loc[["cust_lat", "cust_long"]] = pd.DataFrame(
-        cust_loc["lat_long"].tolist(), index=cust_loc.index
-    )
 
+    cust_loc = pd.read_parquet(r'C:\Users\eelil\OneDrive\Desktop\Capstone\Machine_Learning_Analysis_of_Bank_Fraud\cust_loc.parquet')
 
     df = df.merge(
         cust_loc[["city", "state", "cust_lat", "cust_long"]],
